src/serac/numerics/mesh_utils.cpp

```diff
@@ -167,6 +167,96 @@ std::shared_ptr<mfem::ParMesh> buildCuboidMesh(int elements_in_x, int elements_i
+                                                         {0.0000000000000000, -0.5773502691896258},
+                                                         {0.4082482904638630 * a, -0.4082482904638630 * a},
+                                                         {1.000000000000000, 0.0000000000000000},
+                                                         {0.7071067811865475, 0.7071067811865475},
+                                                         {0.0000000000000000, 1.000000000000000},
+                                                         {-0.707106781186548, 0.7071067811865475},
+                                                         {-1.000000000000000, 0.0000000000000000},
+                                                         {-0.707106781186548, -0.707106781186548},
+                                                         {0.0000000000000000, -1.000000000000000},
+                                                         {0.7071067811865475, -0.707106781186548}};
+
+  static constexpr int elems[num_elems][4] = {{0, 1, 2, 3},   {0, 3, 4, 5},   {0, 5, 6, 7},   {0, 7, 8, 1},
+                                              {1, 9, 10, 2},  {2, 10, 11, 3}, {3, 11, 12, 4}, {4, 12, 13, 5},
+                                              {5, 13, 14, 6}, {6, 14, 15, 7}, {7, 15, 16, 8}, {8, 16, 9, 1}};
+
+  static constexpr int boundary_elems[num_boundary_elements][2] = {{9, 10},  {10, 11}, {11, 12}, {12, 13},
+                                                                   {13, 14}, {14, 15}, {15, 16}, {16, 9}};
+
+  auto mesh = mfem::Mesh(dim, num_vertices, num_elems, num_boundary_elements);
+
+  for (auto vertex : vertices) {
+    mesh.AddVertex(vertex);
+  }
+  for (auto elem : elems) {
+    mesh.AddQuad(elem);
+  }
+  for (auto boundary_elem : boundary_elems) {
+    mesh.AddBdrSegment(boundary_elem);
+  }
+
+  for (int i = 0; i < radial_refinement; i++) {
+    mesh.UniformRefinement();
+  }
+
+  // the coarse mesh is actually a filled octagon
+  // this deforms the vertices slightly to make it
+  // into filled disk instead
+  {
+    int n = mesh.GetNV();
+
+    mfem::Vector new_vertices;
+    mesh.GetVertices(new_vertices);
+    mfem::Vector vertex(dim);
+    for (int i = 0; i < n; i++) {
+      for (int d = 0; d < dim; d++) {
+        vertex(d) = new_vertices[d * n + i];
+      }
+
+      // stretch the octagonal shape into a circle of the appropriate radius
+      double phi = fmod(atan2(vertex(1), vertex(0)) + M_PI, M_PI_4);
+      vertex *= radius * (cos(phi) + (-1.0 + sqrt(2.0)) * sin(phi));
+
+      for (int d = 0; d < dim; d++) {
+        new_vertices[d * n + i] = vertex(d);
+      }
+    }
+    mesh.SetVertices(new_vertices);
+  }
+
+  mfem::Mesh* extruded_mesh = mfem::Extrude2D(&mesh, elements_lengthwise, height);
+
+  auto extruded_pmesh = std::make_shared<mfem::ParMesh>(comm, *extruded_mesh);
+
+  delete extruded_mesh;
```

Comment:
I'd prefer to avoid the manual memory management here, if possible:

```suggestion
  std::unique_ptr<mfem::Mesh> extruded_mesh(mfem::Extrude2D(&mesh, elements_lengthwise, height));

  auto extruded_pmesh = std::make_shared<mfem::ParMesh>(comm, *extruded_mesh);
```